types/share_merging.go

```diff
@@ -26,10 +26,12 @@ func DataFromSquare(eds *rsmt2d.ExtendedDataSquare) (Data, error) {
 
 	// iterate over each row index
 	for x := uint(0); x < originalWidth; x++ {
-		// iterate over each col index
-		for y := uint(0); y < originalWidth; y++ {
+		// iterate over each share in the original data square
+		row := eds.Row(x)
+		originalWidth := eds.Width() / 2
```

Comment:
Why do we need to introduce this variable? In line 17 there is already exactly the same definition.

Reply:
whoops, accidentally declared that twice! thanks for catching this!